src/pages/popup/components/DropdownButton.tsx

```diff
@@ -0,0 +1,240 @@
+  active,
+  openedBookmark,
+  setOpenedBookmark,
+  hoveredBookmark,
+  setHoveredBookmark,
+}: {
+  node: chrome.bookmarks.BookmarkTreeNode;
+  active: boolean;
+  openedBookmark: string | null;
+  setOpenedBookmark: (id: string) => void;
+  hoveredBookmark: string | null;
+  setHoveredBookmark: (id: string) => void;
+}) => {
+  const buttonRef = useRef(null);
+  const openMenu = () => buttonRef?.current?.click();
+  const [openedSubBookmark, setOpenedSubBookmark] = useState<string | undefined>(null);
+  const [hoveredSubBookmark, setHoveredSubBookmark] = useState<string | undefined>(null);
+  const [activeFolder, setActiveFolder] = useState<string | undefined>(null);
+  const [hoveredSubmenu, setHoveredSubmenu] = useState(false);
+  const [openingTimeout, setOpeningTimeout] = useState<ReturnType<typeof setTimeout> | null>(null);
+  const [closingTimeout, setClosingTimeout] = useState<ReturnType<typeof setTimeout> | null>(null);
+  // those refs are necessary to get the current state inside timeouts
+  const openedBookmarkRef = useRef(openedBookmark);
+  openedBookmarkRef.current = openedBookmark;
+  const hoveredBookmarkRef = useRef(hoveredBookmark);
+  hoveredBookmarkRef.current = hoveredBookmark;
+
+  useEffect(() => {
+    if (openedBookmark && openingTimeout) {
+      clearTimeout(openingTimeout);
+      setOpeningTimeout(null);
+    }
+    if (openedBookmark && openedBookmark !== node.id && closingTimeout) {
+      clearTimeout(closingTimeout);
+      setClosingTimeout(null);
+    }
+  }, [openedBookmark]);
+  useEffect(() => {
+    if (!hoveredBookmark || hoveredBookmark !== node.id) {
+      clearTimeout(openingTimeout);
+      setOpeningTimeout(null);
+    }
+  }, [hoveredBookmark]);
+
+  return (
+    <Menu>
+      {({ open, close }) => (
+        <Float
+          composable
+          autoPlacement={{
+            rootBoundary: "document",
+            alignment: "start",
+            allowedPlacements: ["right-start", "right-end"],
+          }}
+          offset={2}
+        >
+          <Float.Reference>
+            <Menu.Button
+              className="w-full"
+              onClick={(event) => {
+                // event.preventDefault();
+                event.stopPropagation();
+              }}
+            >
+              <FolderBookmark
+                ref={buttonRef}
+                node={node}
+                onHover={() => {
+                  if (hoveredBookmark !== node.id) {
+                    setHoveredBookmark(node.id);
+                  }
+                  if (openedBookmark && openedBookmark !== node.id) {
+                    openMenu();
+                  } else if (!openedBookmark) {
+                    setOpeningTimeout(setTimeout(openMenu, 500));
+                  }
+                }}
+                onHoverOut={() => {
+                  if (!hoveredSubmenu) {
+                    setHoveredBookmark(null);
+                    setClosingTimeout(
+                      setTimeout(() => {
```

Comment:
i dont fully get why 0.5s timeout

Reply:
The way chrome bookmarks work is that you click on the bookmark bar's buttons to develop the dropdowns but submenus (subfolders containing other bookmarks) open automatically after a 0.5s timeout. You can test this behavior in chrome bookmarks. Note that they can also be opened more rapidly by clicking on them in this implementation. I think the UX/UI logic behind chrome's choice is that you want to avoid users misclicking on a bookmark when they wanted to go to a subfolder, so all submenu navigation happens without clicking.

Reply:
i see